src/init.cpp

```diff
@@ -908,19 +907,23 @@ void InitParameterInteraction()
             LogPrintf("%s: parameter interaction: -whitebind set -> setting -listen=1\n", __func__);
     }
 
-    if (gArgs.GetBoolArg("-masternode", false)) {
+    if (gArgs.IsArgSet("-masternode")) {
```

Comment:
Not sure if this warning belongs here. `AppInitParameterInteraction()` is probably a better place (next to other `Unsupported argument`s), because `InitParameterInteraction()` is only used to switch the state of one params based on the state of another ones.